Documentation/upgrade.md

```diff
@@ -232,18 +269,42 @@ This is okay as long as the cluster health looks good and all monitors eventuall
 ### Object Storage Daemons (OSDs)
 The OSD pods can be managed in two different ways, depending on how you specified your storage configuration in your [Cluster spec](./cluster-crd.md#cluster-settings).  
 * **Use all nodes:** all storage nodes in the cluster will be managed by a single daemon set.
-Only the one daemon set will need to be edited to update the image version, then each OSD pod will need to be deleted so that a new pod will be created by the daemon set to take its place.
-* **Specify individual nodes:** each storage node specified in the cluster spec will be managed by its own individual replica set.
-Each of these replica sets will need to be edited to update the image version, then each OSD pod will need to be deleted so its replica set will start a new pod on the new version to replace it.
+Only the one daemon set will need to be edited to update the image version
+if upgrading from Rook 0.6.x or older, then each OSD pod will need to be deleted so that a new pod will be created by the daemon set to take its place.
+* **Specify individual nodes:** each storage node specified in the cluster spec will be managed by its own individual deployment.
+Each of these deployment will need to be edited to update the image version, then Kubernetes will start the update and a new pod still start with the new version to replace the old one.
+If upgrading from Rook 0.6.x or older, each OSD pod will be managed by a replica set instead. Each of these replica sets will need to be edited to update the image version, then each OSD pod will need to be deleted so its replica set will start a new pod on the new version to replace it.
+
+If upgrading from v0.6.x or older, you will also need to add a new field `ROOK_CLUSTER_ID` for both cases.
 
 In this example, we are going to walk through the case where `useAllNodes: true` was set in the cluster spec, so there will be a single daemon set managing all the OSD pods.
 
-Let's edit the image version of the pod template in the daemon set:
+#### Upgrading from 0.6.x or lower
+Let's edit the daemon set manifest to update the version and add the new field:
+```bash
+kubectl -n rook edit daemonset rook-ceph-osd
+```
+
+First update the image version to the new version of `rook/rook:master` as shown in the following snippet:
+```
+        image: rook/rook:master
```

Comment:
seems like the `set image` command is simpler than trying to edit the spec

Reply:
It is easier. But we have to edit the spec anyways to add the `ROOK_CLUSTER_ID`. So might as well do it all together.

Reply:
why do you show the `imagePullPolicy` and `name`? it gives the impression that we need to edit them too.